main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	set "github.com/deckarep/golang-set"
+	"github.com/llgcode/draw2d/draw2dimg"
+)
+
+type LRTuple struct {
+	dir  float64
+	dist int
+}
+
+type Tuple struct {
+	rad float64
+	gx  int
+	gy  int
+}
+
+var ArrInput []LRTuple
+var xyInput []Tuple
+var iInput []Tuple
+var xyOutput []interface{}
+
+func printErr(err error) {
```

Comment:
This is a bad idea because this encourages you to just print the error and move on. You'll see why this is a problem in a second...